src/Controls/src/Core/HandlerImpl/VisualElement/VisualElement.Impl.cs

```diff
@@ -306,21 +306,18 @@ static double EnsurePositive(double value)
 
 		void NotifyShadowChanges()
 		{
-			if (Shadow != null)
+			var shadow = Shadow;
+
+			if (shadow is not null)
 			{
-				Shadow.Parent = this;
-				Shadow.PropertyChanged += OnShadowChanged;
+				var proxy = _shadowProxy ??= new();
+				proxy.Subscribe(shadow, (sender, e) => OnPropertyChanged(nameof(Shadow)));
```

Comment:
As seen in: https://github.com/dotnet/maui/pull/13997

I think you can change this to:
```csharp
_shadowHandler ??= (sender, e) => OnPropertyChanged(nameof(Shadow));
_shadowProxy ??= new();
proxy.Subscribe(shadow, _shadowHandler);
```
Where `_shadowHandler` will hold onto the `EventHandler` and make this test pass: https://github.com/dotnet/maui/pull/13960/files#r1139276236

Reply:
Ok, just so that people get why we need the extra local var, we are not trying to fix a thread race condition, but a memory race condition in which the GC kicks ins that is why the code is confusing. There are not thread race conditions because everything should be happening in the main/ui thread. Is just that the *fix* is to use a local var to keep the ref until we don't longer need it.

Reply:
Right, the test was not passing in that case.
Applied changes to save the EventHandlers in a member field.
![image](https://user-images.githubusercontent.com/6755973/225882052-41ca8a94-1969-4bde-bf14-c3e9d7c7db8b.png)